### ISSUE TITLE
Possibility to use alternatives to colmap?

### ISSUE THREAD

**Issue by Shubhra22** [ArthurBrussee/brush#11]
I was wondering if it will be use camera data from iphone(something like record3d) instead of colmap? If not supported as this moment, any directions you can probably provide if I were to DIY? Thank you for the great work

**Comment by ArthurBrussee** [comment]
yes that'd be good! I have no experience with record3d but looks like it might work.

If you'd like to give it a shot, I've cleaned up the code in brush-dataset a bit (https://github.com/ArthurBrussee/brush/commit/ad702606e41b0e3b23aecfe4547ed965accd28d7). You will need to add a new format there, which extracts:
- The images.
- The camera position & rotation & focal for each image

You can construct one "view" with the SceneView struct, which holds an image and a camera. A collection of these is a Scene. A dataset holds a training and potentially an evalution Scene.

Datastes are streamed in, so there is some tricky business around async streams. The idea is to send occasional updates when new images are loaded in.

Lmk if you get stuck on anything!

**Comment by lucas-vitrus** [comment]
I'm testing now with Glomap (which is 1:1 compatible with Colmap output, but 100x faster). I will add here if I can get it working

**Comment by lucas-vitrus** [comment]
Glomap works perfectly! 

Just guarantee you have `/images` and `/sparse` into the `.zip` root.

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/0949e2b4-65bf-4670-82ed-3e97683624d4">

**Comment by ArthurBrussee** [comment]
Nice that's great to hear! If you're happy to, some info in the readme about setting up COLMAP / glomap would be very helpful.

**Comment by Shubhra22** [comment]
I was able to extract transform matrix with AR foundation, can you tell me what is camera_angle_x and rotation value in the json represents? 
![image](https://github.com/user-attachments/assets/05219456-c3c1-4f57-a808-22e018c6dbe1)

**Comment by poly2it** [comment]
> Glomap works perfectly!
> 
> Just guarantee you have `/images` and `/sparse` into the `.zip` root.
> <img alt="image" width="1080" src="https://private-user-images.githubusercontent.com/133435610/385621821-0949e2b4-65bf-4670-82ed-3e97683624d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA3NjYzNDYsIm5iZiI6MTc0MDc2NjA0NiwicGF0aCI6Ii8xMzM0MzU2MTAvMzg1NjIxODIxLTA5NDllMmI0LTY1YmYtNDY3MC04MmVkLTNlOTc2ODM2MjRkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyOFQxODA3MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yY2UzZWZhZmFmZmY3MmJlMTdlOTM4MDA0MDcxMjE3OTNmZGI2ODYxNDNjMmM4NDg5NGMxMTFlNWUzNjM1MzExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X4dYGmjgBh9nQdqEXxHPxlGD6W3zurEx5mHztNQ5CnY">

@lucas-vitrus I am not able to get GLOMAP data to load. Are you still able to? I'm running latest commits on GLOMAP and Brush.

**Comment by poly2it** [comment]
I get a huge volatile blob.

**Comment by poly2it** [comment]
![Image](https://github.com/user-attachments/assets/17a9d7a2-5721-4909-aa01-3cd1cafac752)

**Comment by poly2it** [comment]
#119

**Comment by ArthurBrussee** [comment]
> I was able to extract transform matrix with AR foundation, can you tell me what is camera_angle_x and rotation value in the json represents in the transforms_train.json? Also what is the difference between the transforms_train.json and transforms_val.json ![image](https://github.com/user-attachments/assets/05219456-c3c1-4f57-a808-22e018c6dbe1)

Hey! Sorry I missed this earlier... If it's still relevant/for other people:

-camera_angle_x is the horizontal FOV in radians. Vertical fov can be specified with camera_angle_y, or deduced automatically from the resoltuion.
- transforms_train is for images that are trained on. transforms_val are for images that are used for evaluation.
